ui/server.py

```diff
@@ -316,18 +318,63 @@ def get_repos():
             if repo['url'] == metadata[0]:
                 repo['total'] = metadata[1]  # Total number of discoveries
                 repo['TP'] = metadata[2]  # Number of new discoveries
+                repo['FP'] = metadata[3]  # Number of true positive discoveries
+                # Number of addressing discoveries
+                repo['addressing'] = metadata[4]
+                # Number of irrelevant discoveries
+                repo['not_relevant'] = metadata[5]
+                repo['fixed'] = metadata[6]  # Number of fixed discoveries
                 break  # We found the repo, no need to check next metadata
             else:
                 # The repo was scanned and doesn't have discoveries
                 repo['total'] = 0
                 repo['TP'] = 0
+                repo['FP'] = 0
+                repo['addressing'] = 0
+                repo['not_relevant'] = 0
+                repo['fixed'] = 0
```

Comment:
Are these items re-assigned at every loop? Maybe it's better to move them above the `for`, as an initialisation for the repo

Reply:
Do you suggest having the initialization of the fields before traversing the list of metadata, like this?
```python
    for repo in repos:
        # Init repo
        repo['total'] = 0
        repo['TP'] = 0
        repo['FP'] = 0
        repo['addressing'] = 0
        repo['not_relevant'] = 0
        repo['fixed'] = 0
        for metadata in repos_metadata:
            if repo['url'] == metadata[0]:
                repo['total'] = metadata[1]  # Total number of discoveries
                repo['TP'] = metadata[2]  # Number of new discoveries
                repo['FP'] = metadata[3]  # Number of true positive discoveries
                # Number of addressing discoveries
                repo['addressing'] = metadata[4]
                # Number of irrelevant discoveries
                repo['not_relevant'] = metadata[5]
                repo['fixed'] = metadata[6]  # Number of fixed discoveries
                break  # We found the repo, no need to check next metadata
```

Reply:
yes, I think it's better so to not reassign the same values at every loop, even if the result doesn't change